internal/server/statvar/statvar_hierarchy_util.go

```diff
@@ -164,45 +171,165 @@ func BuildStatVarSearchIndex(
 // A BleveDocument models a document by the bleve index.
 // Currently we index stat vars and treat them as documents.
 type BleveDocument struct {
+	// Id of the statvar.
+	Id string `index:"false"`
 	// Title of the document. For a statvar this will be the DisplayName.
-	Title string
+	Title     string `json:"t"`
+	KeyText   string `json:"k" store:"false"`
+	ValueText string `json:"v" store:"false"`
 	// A key value pairs string describing the properties of a stat var.
-	KeyValueText string
+	KeyValueText string `json:"kv" store:"false"`
+	// Searchable name of the statvar.
+	SearchName string `json:"sn" store:"false"`
+	// Number of constraints associated to a statvar.
+	NumConstraints int32 `json:"nc" store:"false"`
+	// Number of title tokens, the fewer the more generic a statvar.
+	NumTitleTokens int32 `json:"nt" store:"false"`
 }
 
-// BuildBleveIndex builds the bleve search index for all the stat vars.
-func BuildBleveIndex(
-	rawSvg map[string]*pb.StatVarGroupNode,
-) (bleve.Index, error) {
-	defer util.TimeTrack(time.Now(), "BuildBleveIndex")
-	indexMapping := bleve.NewIndexMapping()
-
+func buildBleveDocumentMapping() *mapping.DocumentMapping {
 	documentMapping := bleve.NewDocumentMapping()
-	indexMapping.AddDocumentMapping("Document", documentMapping)
+	documentMapping.Dynamic = false
```

Comment:
what is "Dynamic"?

Reply:
Also feels this function needs more comments on what it does.

Reply:
Added some comments and simplified code.